specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/stable/2018-06-01/QueryPerformanceInsights.json

```diff
@@ -75,6 +75,305 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/queryTexts/{queryId}": {
+      "get": {
+        "tags": [
+          "QueryTexts"
+        ],
+        "operationId": "QueryTexts_Get",
+        "x-ms-examples": {
+          "QueryTextsGet": {
+            "$ref": "./examples/QueryTextsGet.json"
+          }
+        },
+        "description": "Retrieve the Query-Store query texts for the queryId.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/QueryIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/QueryText"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforPostgreSQL/servers/{serverName}/queryTexts": {
+      "get": {
+        "tags": [
+          "QueryTexts"
+        ],
+        "operationId": "QueryTexts_ListByServer",
+        "x-ms-examples": {
+          "QueryTextsListByServer": {
+            "$ref": "./examples/QueryTextsListByServer.json"
+          }
+        },
+        "description": "Retrieve the Query-Store query texts for specified queryIds.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "queryIds",
```

Comment:
GETs should not have required query parameters.  If its not possible to satisfy this GET collection call without input then it should be a POST action like "listQueryTexts"

Reply:
This is an existing API and cannot be changed, similar to [this](https://github.com/Azure/azure-rest-api-specs/blob/412750c48965135973454b582a81901c10529d5b/specification/mysql/resource-manager/Microsoft.DBforMySQL/stable/2018-06-01/QueryPerformanceInsights.json#L87)